Minor changes to Zone11 msgids

marc-vanderwal opened the pull request:
## Purpose

This PR brings the msgids defined in the specification for Zone11 closer to what would be implemented in Engine.

## Context

See https://github.com/zonemaster/zonemaster-engine/issues/1328.

## Changes

Slight changes to the msgids in Zone11.

## How to test this PR

Not applicable.

marc-vanderwal committed:
Minor changes to Zone11 msgids

Bring the msgids closer to what would be implemented in Engine.

marc-vanderwal commented on docs/public/specifications/tests/Zone-TP/zone11.md:
```diff
@@ -41,13 +41,13 @@ server.
+Z11_INCONSISTENT_SPF_POLICIES    | WARNING |              | The zone publishes different SPF policies on different name servers.
+Z11_DIFFERENT_SPF_POLICIES_FOUND | NOTICE  | ns_ip_list   | The following name servers returned the same SPF policy, but other name servers returned a different policy. Name servers: {ns_ip_list}.
+Z11_NO_SPF_FOUND                 | NOTICE  |              | The zone does not publish an SPF policy.
+Z11_SPF1_MULTIPLE_RECORDS        | ERROR   | ns_ip_list   | The following name servers returned more than one SPF policy. Name servers: {ns_ip_list}.
+Z11_SPF1_SYNTAX_ERROR            | ERROR   | ns_ip_list   | The zone’s SPF policy has a syntax error. Policy retrieved from the following nameservers: {ns_ip_list}.
+Z11_SPF1_SYNTAX_OK               | INFO    |              | The zone’s SPF 1 policy has correct syntax.
```
Renaming the message tags will be too disruptive, in my opinion. I suggest we leave them be.

marc-vanderwal commented on docs/public/specifications/tests/Zone-TP/zone11.md:
```diff
@@ -41,13 +41,13 @@ server.
+Z11_NO_SPF_FOUND                 | NOTICE  |              | The zone does not publish an SPF policy.
+Z11_SPF1_MULTIPLE_RECORDS        | ERROR   | ns_ip_list   | The following name servers returned more than one SPF policy. Name servers: {ns_ip_list}.
+Z11_SPF1_SYNTAX_ERROR            | ERROR   | ns_ip_list   | The zone’s SPF policy has a syntax error. Policy retrieved from the following nameservers: {ns_ip_list}.
+Z11_SPF1_SYNTAX_OK               | INFO    |              | The zone’s SPF 1 policy has correct syntax.
```
Indeed, thanks for spotting that!

matsduf commented on docs/public/specifications/tests/Zone-TP/zone11.md:
```diff
@@ -41,13 +41,13 @@ server.
+Z11_DIFFERENT_SPF_POLICIES_FOUND | NOTICE  | ns_ip_list   | The following name servers returned the same SPF policy, but other name servers returned a different policy. Name servers: {ns_ip_list}.
+Z11_NO_SPF_FOUND                 | NOTICE  |              | The zone does not publish an SPF policy.
+Z11_SPF1_MULTIPLE_RECORDS        | ERROR   | ns_ip_list   | The following name servers returned more than one SPF policy. Name servers: {ns_ip_list}.
+Z11_SPF1_SYNTAX_ERROR            | ERROR   | ns_ip_list   | The zone’s SPF policy has a syntax error. Policy retrieved from the following nameservers: {ns_ip_list}.
+Z11_SPF1_SYNTAX_OK               | INFO    |              | The zone’s SPF policy has correct syntax.
+Z11_UNABLE_TO_CHECK_FOR_SPF      | ERROR   |              | None of the zone’s name servers responded with an authoritative response to queries for SPF policies.
```
To be strict, the policy is a feature of a name (mail domain) not a zone. In a zone you could have multiple SPF policies for different mail domains, can't you. E.g. in the zone `test.xa` you could have one for `afnic.test.xa` and another for `iis.test.xa` given that those are used for mail addresses, e.g. `info@afnic.test.xa` and `info@iis.test.xa`, respectively. Zonemaster only tests the SPF policy of the mail domain equal to the zone apex, but that is still just a name.

I suggest something like "The SPF policy of the zone apex (...)"

(I also put the same comment in https://github.com/zonemaster/zonemaster-engine/pull/1348#pullrequestreview-2069648648)

matsduf replied:
Or maybe something like "The SPF policy of {domain} has (...)" where "{domain}" is the name of apex. It will probably be easier to understand than my first suggestion,

marc-vanderwal replied:
I’ll do that. However, adding the `{domain}` argument will require a database migration script in order not to break existing messages, because they do not have any arguments.

matsduf replied:
This is something that we have not considered in the past, but it is not irrelevant. However, if we have that requirement it will be hard to update the messages in the future. When we remove a tag (or rename) it will affect even more.

marc-vanderwal replied:
Yes, database migrations are annoying, but I have another PR open that could use a similar script for the same reason. We could piggyback on that one if we want.

marc-vanderwal committed:
Avoid referring to “the zone’s SPF policy”

Technically, any domain, not just zones, can have SPF policies.
Zonemaster will only fetch the domain under test’s SPF policy. This
means that language referring to “the zone’s SPF policy” is technically
incorrect.

This commit updates the following messages tags to remove such language:

 * Z11_INCONSISTENT_SPF_POLICIES;
 * Z11_NO_SPF_FOUND;
 * Z11_SPF1_SYNTAX_ERROR;
 * Z11_SPF1_SYNTAX_OK.

The three last tags in the list get a new argument, {domain}, which is
necessary for the msgid to stay meaningful and user-friendly. The
alternative would be to say something like “the SPF policy at the zone’s
apex”, but “the SPF policy at domain.example” is much better. This is,
however, a change that should ideally be addressed by a database
migration script so that old tests stay meaningful.

marc-vanderwal commented:
I’ve changed a few more msgids to completely remove language such as “the zone’s SPF policy” in the messages this test case generates. @matsduf, what do you think about the new messages?
